tests.rkt

```diff
@@ -179,8 +179,8 @@
   '(#s(Ans (_.0) ((=/= ((_.0 12)) ((_.0 18)))))))
 
 (test 'diseq-5
-  (run* (p q r s) (=/= (list p p) (list r s)))
-  '(#s(Ans (_.0 _.1 _.2 _.3) ((=/= ((_.0 _.3) (_.2 _.3)))))))
+  (run* (p r s) (=/= (list p p) (list r s)))
+  '(#s(Ans (_.0 _.1 _.2) ((=/= ((_.1 _.2) (_.0 _.2)))))))
```

Comment:
This is a sign that `term<?` is not working properly, since the character `0` is less than `1`.  If you fix the `#t` and `pair` cases of `term<?`, that should fix this problem.